bundles/org.openhab.binding.venstarthermostat/src/main/java/org/openhab/binding/venstarthermostat/internal/discovery/VenstarThermostatDiscoveryService.java

```diff
@@ -0,0 +1,248 @@
+     * @throws IOException
+     * @throws SocketException
+     * @throws UnsupportedEncodingException
+     */
+    private MulticastSocket sendDiscoveryBroacast(NetworkInterface ni)
+            throws UnknownHostException, SocketException, UnsupportedEncodingException {
+        InetAddress m = InetAddress.getByName("239.255.255.250");
+        final int port = 1900;
+
+        log.trace("Considering {}", ni.getName());
+        try {
+            if (!ni.isUp() || !ni.supportsMulticast()) {
+                log.trace("skipping interface {}", ni.getName());
+                return null;
+            }
+
+            Enumeration<InetAddress> addrs = ni.getInetAddresses();
+            InetAddress a = null;
+            while (addrs.hasMoreElements()) {
+                a = addrs.nextElement();
+                if (a instanceof Inet4Address) {
+                    break;
+                } else {
+                    a = null;
+                }
+            }
+            if (a == null) {
+                log.trace("no ipv4 address on " + ni.getName());
+                return null;
+            }
+
+            // for whatever reason, the venstar thermostat responses will not be seen
+            // if we bind this socket to a particular address.
+            // this seems to be okay on linux systems, but osx apparently prefers ipv6, so this
+            // prevents responses from being received unless the ipv4 stack is given preference.
+            MulticastSocket socket = new MulticastSocket(new InetSocketAddress(port));
+            socket.setSoTimeout(2000);
+            socket.setReuseAddress(true);
+            socket.setNetworkInterface(ni);
+            socket.joinGroup(m);
+
+            log.trace("Joined UPnP Multicast group on Interface: " + ni.getName());
+            byte[] requestMessage = COLOR_TOUCH_DISCOVERY_MESSAGE.getBytes("UTF-8");
+            DatagramPacket datagramPacket = new DatagramPacket(requestMessage, requestMessage.length, m, port);
+            socket.send(datagramPacket);
+            return socket;
+        } catch (IOException e) {
+            log.trace("got ioexception: " + e.getMessage());
+        }
+
+        return null;
+    }
+
+    /**
+     * Scans all messages that arrive on the socket and scans them for the
+     * search keywords. The search is not case sensitive.
+     *
+     * @param socket
+     *            The socket where the answers arrive.
+     * @param keywords
+     *            The keywords to be searched for.
+     * @return
+     * @throws IOException
+     */
+    private void scanResposesForKeywords(MulticastSocket socket, String... keywords) throws IOException {
+        // In the worst case a SocketTimeoutException raises
+        do {
+            byte[] rxbuf = new byte[8192];
+            DatagramPacket packet = new DatagramPacket(rxbuf, rxbuf.length);
+            try {
+                socket.receive(packet);
+            } catch (Exception e) {
```

Comment:
Can this be `IOException` ?
```suggestion
            } catch (IOException e) {
```